src/t8_cmesh/t8_cmesh_readmshfile.cxx

```diff
@@ -1729,6 +1751,43 @@ t8_cmesh_msh_file_find_neighbors (t8_cmesh_t cmesh,
 /* This part should be callable from C */
 T8_EXTERN_C_BEGIN ();
 
+/* This is a helper function to properly register the 
+ * geometries for the cmesh created in t8_cmesh_from_msh_file.
+ * It should be called by all processes of the cmesh.
+ * Returns 1 on success, 0 on OCC usage error: use_occ_geometry true, but occ not linked.
+ * The linear_geometry pointer will point to the newly created linear geometry.
+ * The occ_geometry pointer will point to the newly created occ geometry, or to NULL if
+ * no occ geometry is used.
+ */
+static int
+t8_cmesh_from_msh_file_register_geometries (t8_cmesh_t cmesh,
+                                            const int use_occ_geometry,
+                                            const int dim,
+                                            const char *fileprefix,
+                                            const t8_geometry_c
+                                            **linear_geometry,
+                                            const t8_geometry_c
+                                            **occ_geometry)
+{
+
+  const t8_geometry_c *linear_geom = new t8_geometry_linear (dim);
+  /* Register linear geometry */
+  t8_cmesh_register_geometry (cmesh, linear_geom);
+  *linear_geometry = linear_geom;
+  if (use_occ_geometry) {
+#if T8_WITH_OCC
+    const t8_geometry_c *occ_geom =
+      t8_geometry_occ_new (dim, fileprefix, "brep_geometry");
```

Comment:
string / char* handling was never my strong suite. Are string literals existing beyond the scope where they are defined? Here, a `const char*` is expected and simply copied to the geometry object, but for me it is not clear that the underlying memory containing the chars continues to exist.

Reply:
I am also not very familiar with char* handling, but my short google search suggested, that a literal is handled like a char* in read-only memory. Therefore, it is not changeable, but it works.